src/debug/Stream.h

```diff
@@ -37,10 +37,11 @@
 #define MAX_DEBUG_SECTIONS 100
 
 /* defined names for Debug Levels */
-#define DBG_CRITICAL    0   /**< critical messages always shown when they occur */
-#define DBG_IMPORTANT   1   /**< important messages always shown when their section is being checked */
-/* levels 2-8 are still being discussed amongst the developers */
-#define DBG_DATA    9   /**< output is a large data dump only necessary for advanced debugging */
+#define DBG_CRITICAL   0 /**< critical messages always shown when they occur */
+#define DBG_IMPORTANT  1 /**< important messages always shown when their section is being checked */
+#define DBG_PROTOCOL   2 /**< show flow of protocol messages. Excluding message content/payload which is in DBG_DATA. */
+/* levels 3-8 are still being discussed amongst the developers */
+#define DBG_DATA       9 /**< output is a large data dump only necessary for advanced debugging */
 
```

Comment:
If we describe/define `DBG_PROTOCOL` like this, we will need to change 400+ level-2 debugs() messages (that currently do not show protocol messages) to use debug level 3 or higher. Will you do that at the end of this PR work, once all other issues are resolved?

Otherwise, we need to adjust `DBG_PROTOCOL` description/definition to include those level-2 messages.

Reply:
> Will you do that at the end of this PR work, once all other issues are resolved?

I do not intend to do that myself, though I would not object to anyone else doing so.

> 
> Otherwise, we need to adjust `DBG_PROTOCOL` description/definition to include those level-2 messages.

IMO we could define a second label for section 2 of non-protocol sections. This label is not about a severity level like CRITICAL/IMPORTANT were.

Reply:
To address this change request, we need to make a choice:

1. Reserve level-2 debugging for protocol header dumps and such. Add `DBG_PROTOCOL`. Deprecate level-2 use for other purposes. Get rid of legacy uses.
2. Invent a new model/meaning for the debugs() level and/or section parameters to accommodate the proposed "protocol message debugging" use case and especially its [planned](https://github.com/squid-cache/squid/pull/1105#discussion_r931999860) "protocol X message debugging" enhancement. Adjust the PR code accordingly.
3. Do not introduce the new `DBG_PROTOCOL` level name at all. Continue to use anonymous level 2 for message header dumps and other existing use cases.
4. Broaden `DBG_PROTOCOL` level name/meaning to include most current level-2 uses.

You are not excited about item 1 prospects AFAICT, and I share that assessment. I can help with item 2 if that is the direction you favor; it will require non-trivial amounts of work (but hopefully not a lot of additional code changes). I doubt item 4 will work well.

I strongly recommend to start with item 3 because it does not require a lot of work and is fully compatible with future protocol-specific tracing enhancements. With item 3, we can always upgrade to item 2 if/when that becomes necessary without undoing a lot of work.

Reply:
Dropped in [e3a7b48](https://github.com/squid-cache/squid/pull/1105/commits/e3a7b48eb0e00683d6b7284126ccbf260c6ec64b)